src/main/java/com/o19s/es/ltr/ranker/ranklib/DenseProgramaticDataPoint.java

```diff
@@ -53,4 +54,14 @@ public void setFeatureVector(float[] dfVals) {
     public float[] getFeatureVector() {
         return this.fVals;
     }
+
+    @Override
+    public void setFeatureScore(int featureIdx, float score) {
+        this.setFeatureValue(featureIdx+1, score);
```

Comment:
Oh, I see that this list is 1-based... still, the comment would be welcomed as this is unusual enough.